src/mon/NVMeofGwSerialize.h

```diff
@@ -824,22 +847,41 @@ inline void decode(BeaconListener& ls, ceph::buffer::list::const_iterator &bl) {
   DECODE_FINISH(bl);
 }
 
-inline void encode(const BeaconSubsystem& sub,  ceph::bufferlist &bl) {
-  ENCODE_START(1, 1, bl);
+inline void encode(const BeaconSubsystem& sub,  ceph::bufferlist &bl, uint64_t features) {
+  uint8_t version = 1;
+  if (HAVE_FEATURE(features, NVMEOF_BEACON_DIFF)) {
+    version = 2;
+  }
+  // For legacy encoding, skip deleted subsystems to maintain compatibility
+  if (version == 1 &&
+      sub.change_descriptor != subsystem_change_t::SUBSYSTEM_ADDED) {
+    dout(4) << "encode BeaconSubsystem: skipping subsystem " << sub.nqn
+            << " with change_descriptor " << (int)sub.change_descriptor
+            << " in legacy mode" << dendl;
+    return; // Skip encoding this subsystem entirely
+  }
+
+  ENCODE_START(version, version, bl);
   encode(sub.nqn, bl);
+  dout(20) << "encode BeaconSubsystems " << sub.nqn <<" features " <<  features
+           << " version " << (int)version << dendl;
   encode((uint32_t)sub.listeners.size(), bl);
   for (const auto& ls: sub.listeners)
     encode(ls, bl);
   encode((uint32_t)sub.namespaces.size(), bl);
   for (const auto& ns: sub.namespaces)
     encode(ns, bl);
+  if (version >= 2) {
+    encode((uint32_t)sub.change_descriptor, bl);
```

Comment:
This is another example where you are simply appending a field, I don't think you need a conditional encoding here, you could simply unconditionally encode at version 2.